src/static_file_handler.cc

```diff
@@ -105,82 +123,53 @@ RequestHandler::Status StaticFileHandler::Init(const std::string& uri_prefix, co
 RequestHandler::Status StaticFileHandler::HandleRequest(const Request& request, Response* response) {
     std::string file_path = "";
     std::string contents = "";
-    std::string login = "/private/login.html";
+    std::string login = prefix + "/login.html";
     bool redirect = false;
 
-    std::cout << request.raw_request() << std::endl;
+    // Get URI.
+    std::string filename = request.uri();
 
-    if (request.raw_request().find(login) != std::string::npos) {
+    // Check if URI is login page
+    if (filename.find(login) == 0 && login.length() == filename.length()) {
         redirect = true;
     }
 
     // If serving regular static files or the request is about login.html, skip
     if (!username.empty() && !redirect) {
-        time_t cookie_time = 0;
-        std::unordered_map<std::string, time_t>::const_iterator found;
-
-        // Find cookie in the cookie map
-        if (request.cookie() != "") {
-            found = cookie_map.find(request.cookie());
-            if (found != cookie_map.end()) {
-                cookie_time = found->second;
-            }
-        }
-
-        time_t now_seconds;
-        now_seconds = time(NULL);
+        bool cookie_ok = check_cookie(request.cookie(), response);
 
-        if (now_seconds - cookie_time > timeout) {
-            // If no cookie or expired, redirect to login and delete old cookie
-            if (found != cookie_map.end()) {
-                cookie_map.erase(found);
-            }
-
-            response->SetStatus(Response::ResponseCode::FOUND);
-            response->AddHeader("Location", "/private/login.html");
-            response->AddHeader("Set-Cookie", "private=" + request.cookie() + "; expires=Thu, Jan 01 1970 00:00:00 UTC;");
-            response->AddHeader("Content-Type", "text/html");
-            response->AddHeader("Content-Length", "228");
-            get_file("private_files/login.html", &contents);
-            response->SetBody(contents);
-            original_request = request.uri();
+        if (!cookie_ok) {
+            // If cookie is not ok, need to redirect
+            original_uri = filename;
             return RequestHandler::Status::OK;
         }
     }
 
     if (request.method() == "POST" && redirect) {
+        // The body returns: username=USERNAME&password=PASSWORD
         // Extract username and password
         std::string body = request.body();
-        size_t first = body.find("=");
+        size_t first = body.find("=") + 1;
```

Comment:
also i don't think body will be "" either since the method has to be post and has to come from login.html